src/lines_lf.rs

```diff
@@ -47,51 +47,62 @@ pub fn to_byte_idx(text: &str, line_idx: usize) -> usize {
 }
 
 //-------------------------------------------------------------
+const LF: u8 = b'\n';
 
 #[inline(always)]
 fn to_byte_idx_impl<T: ByteChunk>(text: &[u8], line_idx: usize) -> usize {
+    let mut byte_count = 0;
+    let mut lf_count = 0;
+
+    if text.len() <= T::SIZE {
+        // Bypass the more complex routine for short strings, where the
+        // complexity hurts performance.
+        for (i, byte) in text.iter().enumerate() {
+            if lf_count == line_idx {
+                return i;
+            }
+            if *byte == LF {
+                lf_count += 1;
+            }
+        }
+        return text.len();
+    }
     // Get `middle` so we can do more efficient chunk-based counting.
     // We can't use this to get `end`, however, because the start index of
     // `end` actually depends on the accumulating line counts during the
     // counting process.
     let (start, middle, _) = unsafe { text.align_to::<T>() };
 
-    let mut byte_count = 0;
-    let mut lf_count = 0;
-
     // Take care of any unaligned bytes at the beginning.
     for byte in start.iter() {
         if lf_count == line_idx {
-            break;
+            return byte_count;
+        }
+        if *byte == LF {
+            lf_count += 1;
```

Comment:
5% faster if the index is near the start of the string. I am not sure what qualifies as "noticeable". Also this won't show up in the current benchmarks because they only test the case where is index is at the very end.

    lines_lf::to_byte_idx_start/lines_100
                            time:   [5.8930 ns 5.8941 ns 5.8955 ns]
                            thrpt:  [17.851 GiB/s 17.855 GiB/s 17.858 GiB/s]
                     change:
                            time:   [-5.1400% -5.0178% -4.9063%] (p = 0.00 < 0.05)
                            thrpt:  [+5.1595% +5.2829% +5.4186%]
                            Performance has improved.
    lines_lf::to_byte_idx_start/lines_1000
                            time:   [5.8924 ns 5.8940 ns 5.8963 ns]
                            thrpt:  [178.48 GiB/s 178.55 GiB/s 178.60 GiB/s]
                     change:
                            time:   [-5.1775% -4.9822% -4.7803%] (p = 0.00 < 0.05)
                            thrpt:  [+5.0203% +5.2435% +5.4602%]
                            Performance has improved.
    lines_lf::to_byte_idx_start/lines_10000
                            time:   [5.8934 ns 5.8978 ns 5.9043 ns]
                            thrpt:  [1782.4 GiB/s 1784.4 GiB/s 1785.7 GiB/s]
                     change:
                            time:   [-5.1757% -5.0013% -4.8254%] (p = 0.00 < 0.05)
                            thrpt:  [+5.0700% +5.2646% +5.4582%]
                            Performance has improved.

Reply:
I think that counts as notable, yeah.